src/mc-fle2-insert-update-payload-v2.c

```diff
@@ -444,18 +444,22 @@ SERIALIZE_TEXT_TOKEN_SET_FOR_TYPE_IMPL(Prefix)
         const char *index_string = NULL;                                                                               \
         char storage[16];                                                                                              \
         uint32_t index = 0;                                                                                            \
-                                                                                                                       \
-        for (size_t i = 0; i < array->len; i++) {                                                                      \
-            mc_Text##Type##TokenSet_t ts = _mc_array_index(array, mc_Text##Type##TokenSet_t, i);                       \
+        uint32_t limit = (array->len > UINT32_MAX) ? UINT32_MAX : (uint32_t)array->len;                                \
+        while (index < limit) {                                                                                        \
+            mc_Text##Type##TokenSet_t ts = _mc_array_index(array, mc_Text##Type##TokenSet_t, index);                   \
             bson_uint32_to_string(index, &index_string, storage, sizeof(storage));                                     \
                                                                                                                        \
             if (!_fle2_serialize_Text##Type##TokenSet(&arr_bson, index_string, &ts)) {                                 \
                 return false;                                                                                          \
             }                                                                                                          \
-            if (index == UINT32_MAX) {                                                                                 \
-                break;                                                                                                 \
-            }                                                                                                          \
             index++;                                                                                                   \
+        }                                                                                                              \
+        if (index == UINT32_MAX && array->len > UINT32_MAX) {                                                          \
```

Comment:
Not sure if I understand -- Why are we appending one extra element in this case?

Reply:
bc the loop above it is not inclusive of index at UINT32_MAX if the array length exceeds that.  highly unlikely, but this is the same logic for range edge tokens in `mc_FLE2InsertUpdatePayloadV2_serializeForRange()`.

Reply:
My question is - if we are ignoring all the elements beyond index `UINT32_MAX`, why not also ignore the element at `UINT32_MAX`? Why do we have an artificial limit of `UINT32_MAX + 1` elements?

Reply:
I agree its arbitrary and we don't need the extra code. Removed.